src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleBinaryOperator.cs

```diff
@@ -0,0 +1,414 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class LocalRewriter
+    {
+        /// <summary>
+        /// Rewrite `GetTuple() == (1, 2)` to `tuple.Item1 == 1 &amp;&amp; tuple.Item2 == 2`.
+        /// Also supports the != operator, nullable and nested tuples.
+        ///
+        /// Note that all the side-effects for visible expressions are evaluated first and from left to right. The initialization phase
+        /// contains side-effects for:
+        /// - single elements in tuple literals, like `a` in `(a, ...) == (...)` for example
+        /// - nested expressions that aren't tuple literals, like `GetTuple()` in `(..., GetTuple()) == (..., (..., ...))`
+        /// On the other hand, `Item1` and `Item2` of `GetTuple()` are not saved as part of the initialization phase of `GetTuple() == (..., ...)`
+        ///
+        /// Element-wise conversions occur late, together with the element-wise comparisons. They may not be evaluated.
+        /// </summary>
+        /// There are 2 cases:
+        /// - tuple literal cannot be null and it produces as many temps as elements
+        /// - tuple types or nullable tuple types produce a single temp (and no further temps are initialized)
+        public override BoundNode VisitTupleBinaryOperator(BoundTupleBinaryOperator node)
+        {
+            var boolType = node.Type; // we can re-use the bool type
+            var initialEffectsAndTemps = TupleOperatorSideEffectsAndTemps.GetInstance();
+
+            var returnValue = RewriteTupleOperator(node.Operators, node.Left, node.Right, boolType, initialEffectsAndTemps, node.OperatorKind, inLeftLiteral: true, inRightLiteral: true);
```

Comment:
>inLeftLiteral: true [](start = 143, length = 19)

I do not understand why are we unconditionally passing true here, the operand doesn't have to be a literal.

Reply:
No longer applicable.

---
In reply to: [171080574](https://github.com/dotnet/roslyn/pull/23356#discussion_r171080574) [](ancestors = 171080574)